scipy/optimize/tests/test_optimize.py

```diff
@@ -502,6 +502,18 @@ def test_ncg_hessp(self):
                         atol=1e-6, rtol=1e-7)
 
 
+def test_maxfev_test():
+    def cost(x):
+        return np.random.rand(1) * 1000  # never converged problem
+
+    for imaxfev in [1, 10, 50]:
+        for method in ['Powell']:  # TODO: extend to more methods
```

Comment:
What do you mean with TODO? Do you propose to do this for other methods? If yes, did you already try?

Reply:
As reported in #12271, SLSQP might have a similar issue. Also, I think all optimizers should be checked by this test. 
> did you already try?

Not yet. I will try it next PR.

Reply:
ok let's keep this for a follow up.